src/emacs/lean-server.el

```diff
@@ -44,8 +47,9 @@
        (setf (lean-server-session-tasks sess) res)
        (lean-server-notify-tasks-changed sess old-tasks)))
     ("error"
-     (message "error: %s" (plist-get res :message))
-     ; TODO(gabriel): maybe even add the error as a message
+     (when lean-server-print-errors-p
+       (message "error: %s" (plist-get res :message)))
```

Comment:
> Yes, the hole commands were reporting errors when run outside of a hole.

No, that's fine.  But the errors should cause the error callback to be called (the last argument to `lean-server-send-command`).  This line should only be executed for errors that are not related to a request.